src/Common/Uuid/UuidRegistry.php

```diff
@@ -49,9 +51,19 @@ public function createUuid()
                 exit;
             }
 
-            // Create uuid
-            $uuid4 = Uuid::uuid4();
-            $uuid = $uuid4->toString();
+            // Create uuid using the Timestamp-first COMB Codec, so can use for primary keys
+            //  (since first part is timestamp, it is naturally ordered)
+            //  reference:
+            //    https://uuid.ramsey.dev/en/latest/customize/timestamp-first-comb-codec.html#customize-timestamp-first-comb-codec
+            $factory = new UuidFactory();
+            $codec = new TimestampFirstCombCodec($factory->getUuidBuilder());
+            $factory->setCodec($codec);
+            $factory->setRandomGenerator(new CombGenerator(
+                $factory->getRandomGenerator(),
+                $factory->getNumberConverter()
+            ));
+            $timestampFirstComb = $factory->uuid4();
+            $uuid = $timestampFirstComb->getBytes();
```

Comment:
@jesdynf , What are your thoughts on this mechanism for uuid as a primary key. More details on "Timestamp-first COMB Codec" uuid can be found here:
https://uuid.ramsey.dev/en/latest/customize/timestamp-first-comb-codec.html#customize-timestamp-first-comb-codec

Reply:
My review suggests that it's a popular answer. I found a Percona (MySQL consultancy) article that griped about uuid PKs and suggested a hack to fix them that basically boiled down to manually moving timestamp bytes around to mangle the uuid to be timestamp-prefaced.

Reply:
Still looks like still will have 74 bits of randomness (timestamp takes up 48 bits) which should definitely be more than enough.
After some more testing, plan to bring this in since will open door to potentially support offsite support in the distant future (will allow lots of time to do performance tests etc until we ever get to the point where we drop the counter primary key and go to a uuid).

Reply:
cool, the `uuid4` is 128 bits? so would be 80 bits of randomness?

Reply:
I think there are some bits that are used for housekeeping stuff.